packages/next/src/server/lib/incremental-cache/fetch-cache.ts

```diff
@@ -220,6 +229,11 @@ export default class FetchCache implements CacheHandler {
           throw new Error(`invalid cache value`)
         }
 
+        // if new tags were specified, merge those tags to the existing tags
+        if (cached.kind === 'FETCH') {
+          cached.tags = [...new Set([...(cached.tags ?? []), ...(tags ?? [])])]
```

Comment:
This reads weird to me. I'd suggest something simpler, like:
```suggestion
          cached.tags ??= [];
          for (const tag of tags) {
            if (!cached.tags.includes(tag)) {
              cached.tags.push(tag)
            }
          }
```

Reply:
@Ethan-Arrowood That was actually what I had initially, but using `new Set` makes it more efficient. Maybe it's just a little weird with the `?? []`, but I think this is fine.

Reply:
Did you profile this? What is the possible size of `tags`? I think if its not going to be in the thousands of items then I'd prefer readability over big-o faster code.

Reply:
I'd be willing to wager that given a small enough size _N_ of `tags`, the overhead of instantiating a `Set` will be greater than that of a couple array methods.

Reply:
@Ethan-Arrowood Have not profiled this.

Yeah it's likely negligibly slower instantiating a Set with smaller number of tags. I see  that the max number of tags we can have is 95 → https://github.com/vercel/next.js/blob/canary/test/e2e/app-dir/app-static/app/too-many-cache-tags/page.tsx.

I don't see how it's less readable personally, but since it's unlikely that users will have that many tags, I'll update it.